File: mcdowell/python/ch11-01.py
def merge(left, right):
    rightCount = len(right)
    leftCount = len(left) - rightCount


    leftIndex = leftCount - 1
    rightIndex = rightCount - 1

    totalCount = leftCount + rightCount

    counter = 0


    while leftIndex >= 0 and rightIndex >= 0:
        if right[rightIndex] > left[leftIndex]:
            left[totalCount - 1 - counter] = right[rightIndex]

            rightIndex -= 1
        else:
            left[totalCount - 1 - counter] = left[leftIndex]
            leftIndex -= 1

        counter += 1


    while rightIndex >= 0:
        left[totalCount - 1 - counter] = right[rightIndex]
        rightIndex -= 1
        counter += 1


    return left



left = [10, 11, 20, 40, 50, 0, 0, 0, 0, 0, 0]
right = [14, 14, 51, 99, 100, 100]

print merge(left, right)

File: mcdowell/php/ch11-01.php
<?php



function merge($left, $right) {

    $rightCount = count($right);
    $leftCount = count($left) - $rightCount;

    $rightIndex = $rightCount - 1;
    $leftIndex = $leftCount - 1;

    $totalCount = $rightCount + $leftCount;

    $counter = 0;
    while ($rightIndex >= 0 && $leftIndex >= 0) {

        if ($right[$rightIndex] > $left[$leftIndex]) {
            $left[$totalCount - 1 - $counter] = $right[$rightIndex];
            $rightIndex --;

        }
        else {
            $left[$totalCount - 1 - $counter] = $left[$leftIndex];
            $leftIndex --;
        }
        $counter ++;
    }

    while ($rightIndex >= 0) {
        $left[$totalCount - 1 - $counter] = $right[$rightIndex];
        $counter ++;
        $rightIndex --;
    }


    return $left;
}



$left = [11,12,23,24,35,36,47,null,null,null];
$right = [1,24,26];

print_r(merge($left, $right));